pdcupdater/utils.py

```diff
@@ -694,3 +696,23 @@ def inner(*args, **kwargs):
                     time.sleep(interval)
         return inner
     return wrapper
+
+
+def get_token(pdc_api_url, keytab):
+    """
+    Uses kerberos keytab for automatic authentication 
+    and retrieves the token from pdc-updater
+    """
+    if not os.path.exists(keytab):
+        raise OSError("Keytab file not found")
+
+    os.environ['KRB5_CLIENT_KTNAME'] = 'FILE:{0}'.format(keytab)
```

Comment:
I think it would be better to define keytab file and principal name in the config. Using env variable will work with your personal keytab, but principals work a bit differently - you should actually run kinit on the keytab with the principal name to get kerberos auth. This might only work if there's a separate process running kinit